gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -1526,30 +1552,57 @@ bool netCDFLayer::FillVarFromFeature(OGRFeature *poFeature, int nMainDimId,
                              m_poFeatureDefn->GetFieldDefn(i)->GetNameRef());
                     m_aoFieldDesc[i].bHasWarnedAboutTruncation = true;
                 }
-                status = nc_put_var1_text(
-                    m_nLayerCDFId, m_aoFieldDesc[i].nVarId, anIndex, pszVal);
+                if(m_poDS->HasInfiniteRecordDim())
+                {
+                    status = nc_put_var1_text(
+                        m_nLayerCDFId, m_aoFieldDesc[i].nVarId, anIndex, pszVal);
+                }
+                else
+                {
+                    std::shared_ptr<nccfdriver::OGR_SGFS_Transaction> ptr(new nccfdriver::OGR_SGFS_NC_Char_Transaction(m_aoFieldDesc[i].nVarId, pszVal));
+                    m_poDS->FieldScribe.enqueue_transaction(ptr);
+                }
             }
             else
             {
                 size_t anCount[2];
                 anCount[0] = 1;
                 anCount[1] = strlen(pszVal);
                 size_t nWidth = 0;
-                nc_inq_dimlen(m_nLayerCDFId, m_aoFieldDesc[i].nSecDimId,
-                              &nWidth);
+
+                if(m_bLegacyCreateMode)
+                {
+                    nc_inq_dimlen(m_nLayerCDFId, m_aoFieldDesc[i].nSecDimId,
+                                  &nWidth);
+                }
+                else
+                {
+                    nWidth = m_poDS->vcdf.virtualDIDToDim(m_aoFieldDesc[i].nSecDimId).getLen();
+                }
+
                 if( anCount[1] > nWidth )
                 {
-                    if( m_bAutoGrowStrings &&
-                        m_poFeatureDefn->GetFieldDefn(i)->GetWidth() == 0 )
+                    // Always grow the dim if not writing to WKT- it's rather inexpensive in CF-1.8
+
+                    if( (m_bAutoGrowStrings &&
+                         m_poFeatureDefn->GetFieldDefn(i)->GetWidth() == 0) || !m_bLegacyCreateMode)
                     {
                         size_t nNewSize = anCount[1] + anCount[1] / 3;
 
                         CPLDebug("GDAL_netCDF", "Growing %s from %u to %u",
                                  m_poFeatureDefn->GetFieldDefn(i)->GetNameRef(),
                                  static_cast<unsigned>(nWidth),
                                  static_cast<unsigned>(nNewSize));
-                        m_poDS->GrowDim(m_nLayerCDFId,
-                                        m_aoFieldDesc[i].nSecDimId, nNewSize);
+
+                        if(m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation

Reply:
Oh. Well the netCDFVID doesn't actually have a handle on m_poDS inside (and intentionally) so I think this specific one, concerning GrowDim, will be left the same for now.